Src/Particle/AMReX_NeighborList.H

```diff
@@ -334,8 +334,8 @@ public:
         // First pass: count the number of neighbors for each particle
         //---------------------------------------------------------------------------------------------------------
         const size_t np_size  = (num_bin_types > 1) ? np_total : np_real;
-        m_nbor_counts.resize( np_size+1, 0);
-        m_nbor_offsets.resize(np_size+1);
```

Comment:
I believe the iterator start / stop logic [here](https://github.com/WeiqunZhang/amrex/blob/a75649b9ced9418affd7ff89f1ab5da6bb0c060e/Src/Particle/AMReX_NeighborList.H#L158) relies on the offsets array having size `np+1` and its last element being the total size.